types/tampermonkey/index.d.ts

```diff
@@ -531,3 +531,157 @@ declare function GM_setClipboard(
+    addStyle(css: string): Promise<HTMLStyleElement>;
+
+    // Storage
+
+    /** Sets the value of `name` to the storage */
+    setValue(name: string, value: any): Promise<void>;
+
+    /** Gets the value of 'name' from storage */
+    getValue<TValue>(name: string, defaultValue?: TValue): Promise<TValue>;
+
+    /** Deletes 'name' from storage */
+    deleteValue(name: string): Promise<void>;
+
+    /** Lists all names of the storage */
+    listValues(): Promise<string[]>;
+
+    /**
+     * Adds a change listener to the storage and returns the listener ID.
+     * The `remote` argument of the callback function shows whether this value was
+     * modified from the instance of another tab (`true`) or within this script
+     * instance (`false`). Therefore this functionality can be used by scripts of
+     * different browser tabs to communicate with each other.
+     * @param name Name of the observed variable
+     */
+    addValueChangeListener(): Promise<number>;
+
+    /** Removes a change listener by its ID */
+    removeValueChangeListener(listenerId: number): Promise<void>;
+
+    // Resources
+
+    /** Get the content of a predefined `@resource` tag at the script header */
+    getResourceText(name: string): Promise<string>;
+
+    // note that it's `..Url`, not `URL` like the GM_ variant
+    /**
+     * Get the base64 encoded URI of a predefined `@resource` tag at the script
+     * header
+     */
+    getResourceUrl(name: string): Promise<string>;
+
+    // Menu commands
+
+    /**
+     * Register a menu to be displayed at the Tampermonkey menu at pages where this
+     * script runs and returns a menu command ID.
+     */
+    registerMenuCommand(name: string, onClick: () => void, accessKey?: string): Promise<number>;
+    /**
+     *  Unregister a menu command that was previously registered by
+     * `GM_registerMenuCommand` with the given menu command ID.
+     */
+    unregisterMenuCommand(menuCommandId: number): Promise<void>;
+
+    // Requests
+
+    // note the uppercase H in `Http`
+    /**
+     * Makes an xmlHttpRequest
+     *
+     * @throws {Tampermonkey.ErrorResponse}
+     */
+    xmlHttpRequest<TContext = any>(
+        // onload and the like still work
+        details: Tampermonkey.Request<TContext>, // tslint:disable-line:no-unnecessary-generics
+    ): Promise<Tampermonkey.Response<TContext>>;
+
+    // GM_download has two ways but GM.download doesn't
+    /**
+     * Downloads a given URL to the local disk
+     *
+     * @throws {Tampermonkey.DownloadErrorResponse}
+     */
+    download(details: Tampermonkey.DownloadRequest): Promise<void>;
+
+    // Tabs
+
+    /** Saves the tab object to reopen it after a page unload */
+    saveTab(obj: object): Promise<void>;
+
+    /** Gets a object that is persistent as long as this tab is open */
+    getTab(): Promise<any>;
+
+    /** Gets all tab objects as a hash to communicate with other script instances */
+    getTabs(): Promise<{ [tabId: number]: any }>;
```

Comment:
I wonder how a user is supposed to match tabId to a tab

Reply:
I couldn't find any pattern